src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -888,15 +893,14 @@ private void saveHttpResponseHeaders(Map<String, String> headers) {
    * @return session key
    */
   public static byte[] getSessionKey() {
+    return ExchangeServiceBase.binarySecret;
+  }
+  
+  private static byte[] generateSessionKey(){
     // this has to be computed only once.
-    synchronized (ExchangeServiceBase.class) {
-      if (ExchangeServiceBase.binarySecret == null) {
-        Random randomNumberGenerator = new Random();
-        ExchangeServiceBase.binarySecret = new byte[256 / 8];
-        randomNumberGenerator.nextBytes(binarySecret);
-      }
-
-      return ExchangeServiceBase.binarySecret;
-    }
+	Random randomNumberGenerator = new SecureRandom();
+	final byte[] bytes = new byte[256 / 8];
```

Comment:
Since this is an existing issue and is unrelated, we should keep it as is.  If you desire to fix it, then do it as a separate PR.  Let's keep PRs focused on one issue.